lightning-block-sync/src/http.rs

```diff
@@ -207,25 +213,45 @@ impl HttpClient {
 		#[cfg(not(feature = "tokio"))]
 		let mut reader = std::io::BufReader::new(limited_stream);
 
-		macro_rules! read_line { () => { {
-			let mut line = String::new();
-			#[cfg(feature = "tokio")]
-			let bytes_read = reader.read_line(&mut line).await?;
-			#[cfg(not(feature = "tokio"))]
-			let bytes_read = reader.read_line(&mut line)?;
-
-			match bytes_read {
-				0 => None,
-				_ => {
-					// Remove trailing CRLF
-					if line.ends_with('\n') { line.pop(); if line.ends_with('\r') { line.pop(); } }
-					Some(line)
-				},
+		macro_rules! read_line {
+			() => { read_line!(0) };
+			($retry_count: expr) => { {
+				let mut line = String::new();
+				let mut timeout_count: u64 = 0;
+				let bytes_read = loop {
+					#[cfg(feature = "tokio")]
+					let read_res = reader.read_line(&mut line).await;
+					#[cfg(not(feature = "tokio"))]
+					let read_res = reader.read_line(&mut line);
+					match read_res {
+						Ok(bytes_read) => break bytes_read,
+						Err(e) if e.kind() == std::io::ErrorKind::WouldBlock => {
```

Comment:
Any other errors we would want to retry on? Like `Interrupted` docs https://doc.rust-lang.org/std/io/enum.ErrorKind.html#variant.Interrupted seems to indicate that it can probably be retried

Reply:
Hmm, I don't think you can get an Interrupted in a read on Linux and probably OSX. I'm not really sure what it would even mean in practice - usually it would imply another thread someone sent some signal to interrupt, but in a read not so much.

All that said, maybe it makes sense to just retry everything. Another potential error we could see is a Bitcoin Core RPC queue full error, which would present as just a generic HTTP error.